src/core/ConversationModel.cpp

```diff
@@ -316,3 +318,13 @@ int ConversationModel::indexOfIdentifier(MessageId identifier, bool isOutgoing)
     return -1;
 }
 
+void ConversationModel::prune()
+{
+    const int history_limit = 1000;
+    while(messages.size() > history_limit)
+    {
+        beginRemoveRows(QModelIndex(), messages.size()-1, messages.size()-1);
+        messages.removeLast();
+        endRemoveRows();
```

Comment:
Fun Qt fact: a _lot_ of things are recalculated, laid out, possibly rendered, and otherwise inefficiently processed on model change signals.

In this case it doesn't matter, because it will never end up in a position to remove more than one row. Otherwise, I'd insist on changing it to a single `beginRemoveRows(QModelIndex(), history_limit, messages.size()-1)`.

I'm happy either way here.